### ISSUE TITLE
Error while converting MaskRCnn custom trained model

### ISSUE THREAD

**Issue by vishnujakhoria** [openvinotoolkit/openvino#12343]
Hi,

I am trying to convert a custom trained MaskRCNN model. 

I downloaded the model from here:   https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf2_detection_zoo.md

and trained it on custom dataset.

Tensorflow version used for training: 2.6

I used the following command to convert the model:

mo --saved_model_dir "models/gcp_model_maskrcnn_with_masks/export_run2/export/saved_model" --transformations_config "/root/sharedfolder/dent_damage_det/mask_rcnn_support_api_v2.4.json" --tensorflow_object_detection_api_pipeline_config "models/gcp_model_maskrcnn_with_masks/export_run2/export/pipeline.config" --output_dir "models/gcp_model_maskrcnn_with_masks/export_run2/export/openvino_model" --input_shape [1,1024,1024,3] --reverse_input_channels

 

The conversion is giving me this error:

[ ERROR ]  Exception occurred during running replacer "ObjectDetectionAPIPreprocessor2Replacement" (<class 'openvino.tools.mo.front.tf.ObjectDetectionAPI.ObjectDetectionAPIPreprocessor2Replacement'>): The matched sub-graph contains network input node "input_tensor". 
 For more information please refer to Model Optimizer FAQ, question #75. (https://docs.openvino.ai/latest/openvino_docs_MO_DG_prepare_model_Model_Optimizer_FAQ.html?question=75#question-75)

The custom model and pipeline config file is available at this location: 

https://drive.google.com/drive/u/2/folders/1DxeiIgOi8MFWGxm206NzFf5EB_Jok0Ig

Please help me to resolve this issue.

Thanks,
Vishnu

**Comment by zulkifli-halim** [comment]
Hi @vishnujakhoria. I replicated the issue on my end and found a similar error.
```
OpenVINO runtime version:       2022.1.0-7019-cdb9bec7210-releases/2022/1
Model Optimizer version:        2022.1.0-7019-cdb9bec7210-releases/2022/1
[ ERROR ]  Exception occurred during running replacer "ObjectDetectionAPIPreprocessor2Replacement" (<class 'openvino.tools.mo.front.tf.ObjectDetectionAPI.ObjectDetectionAPIPreprocessor2Replacement'>): The matched sub-graph contains network input node "input_tensor".
For more information please refer to Model Optimizer FAQ, question #75. (https://docs.openvino.ai/latest/openvino_docs_MO_DG_prepare_model_Model_Optimizer_FAQ.html?question=75#question-75)
```

This error is due to the transformation config you are using in the MO command that is not compatible with TensorFlow 2.6. We will investigate this further and get back to you as soon as possible.

**Comment by vishnujakhoria** [comment]
Hi @zulkifli-halim did you find any resolution to the issue? Please help me to resolve it. Is there any tool or method to generate transformation config for Tensorflow 2.6?

**Comment by hbalasu1** [comment]
Hi @vishnujakhoria 

I have duplicated your issue, and it seems like a bug. I have escalated this issue to the respective team, and I will inform you once I get a response. 

Thank you 

Ref. 89368

**Comment by hbalasu1** [comment]
Hi @vishnujakhoria 
Thank you for your patience, Here is the resolution for the issue; 

Converting this model requires a custom .json config as the model is modified and node names differ from the public model.
here we attached the config file mask_rcnn_support_api_v2.4_custom_trained.json in
[mask_rcnn_support_api_v2.4_custom_trained.zip](https://github.com/openvinotoolkit/openvino/files/10334246/mask_rcnn_support_api_v2.4_custom_trained.zip) to this ticket which can be used for conversion.
So the command line for conversion is the following:

`mo --saved_model_dir "./MaskRcnn_custom_train/export/saved_model" --transformations_config "mask_rcnn_support_api_v2.4_custom_trained.json" --tensorflow_object_detection_api_pipeline_config "./MaskRcnn_custom_train/export/pipeline.config" --input_shape [1,1024,1024,3] --reverse_input_channels`

There is also a bug in MO which affects model accuracy after conversion. The fix is on review: https://github.com/openvinotoolkit/openvino/pull/14869

**Comment by hbalasu1** [comment]
Hi @vishnujakhoria 
Fix #14869 has been merged. You can git pull the latest master branch and verify, and it should work now. 
Once you verify, please inform us to close the case. 

Thank you

**Comment by hbalasu1** [comment]
Hi @vishnujakhoria 
Since the issue is fixed, I will close this case. if you need further assistance, please feel free to create new case. 

Thank you
